modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-impl/src/main/java/com/liferay/headless/commerce/admin/channel/internal/resource/v1_0/ProductDisplayPageResourceImpl.java

```diff
@@ -0,0 +1,226 @@
+					commerceChannel.getGroupId())
+			).build(),
+			transform(
+				cpDisplayLayoutBaseModelSearchResult.getBaseModels(),
+				cpDisplayLayout -> _toProductDisplayPage(
+					cpDisplayLayout.getCPDisplayLayoutId())));
+	}
+
+	public ProductDisplayPage getProductDisplayPage(Long id) throws Exception {
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.fetchCPDisplayLayout(id);
+
+		if (cpDisplayLayout == null) {
+			throw new NoSuchCPDisplayLayoutException();
+		}
+
+		return _toProductDisplayPage(cpDisplayLayout.getCPDisplayLayoutId());
+	}
+
+	public ProductDisplayPage patchProductDisplayPage(
+			Long id, ProductDisplayPage productDisplayPage)
+		throws Exception {
+
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.fetchCPDisplayLayout(id);
+
+		if (cpDisplayLayout == null) {
+			throw new NoSuchCPDisplayLayoutException();
+		}
+
+		_cpDisplayLayoutService.updateCPDisplayLayout(
+			id,
+			GetterUtil.getLong(
+				productDisplayPage.getProductId(),
+				cpDisplayLayout.getClassPK()),
+			cpDisplayLayout.getLayoutPageTemplateEntryUuid(),
+			GetterUtil.getString(
+				productDisplayPage.getPageUuid(),
+				cpDisplayLayout.getLayoutUuid()));
+
+		return _toProductDisplayPage(id);
+	}
+
+	public ProductDisplayPage
+			postChannelByExternalReferenceCodeProductDisplayPage(
+				String externalReferenceCode,
+				ProductDisplayPage productDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.
+				getCommerceChannelByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		return postChannelProductDisplayPage(
+			commerceChannel.getCommerceChannelId(), productDisplayPage);
+	}
+
+	public ProductDisplayPage postChannelProductDisplayPage(
+			Long id, ProductDisplayPage productDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(id);
+
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.addCPDisplayLayout(
+				commerceChannel.getSiteGroupId(), CPDefinition.class,
+				productDisplayPage.getProductId(),
+				productDisplayPage.getPageTemplateUuid(),
+				productDisplayPage.getPageUuid());
+
+		return _toProductDisplayPage(cpDisplayLayout.getCPDisplayLayoutId());
+	}
+
+	private Map<String, Map<String, String>> _getActions(
+		CPDisplayLayout cpDisplayLayout) {
+
+		return HashMapBuilder.<String, Map<String, String>>put(
+			"delete",
+			addAction(
+				ActionKeys.DELETE, cpDisplayLayout.getCPDisplayLayoutId(),
+				"deleteProductDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
+		).put(
+			"get",
+			addAction(
+				ActionKeys.VIEW, cpDisplayLayout.getCPDisplayLayoutId(),
+				"getProductDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
+		).put(
+			"patch",
+			addAction(
+				ActionKeys.UPDATE, cpDisplayLayout.getCPDisplayLayoutId(),
+				"patchProductDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
```

Comment:
Let's use directly the ModelResourcePermission of the commerceCatalog here instead of `_cpDisplayLayoutModelResourcePermission`